lms/djangoapps/commerce/api/v1/serializers.py

```diff
@@ -26,11 +29,31 @@ class CourseSerializer(serializers.Serializer):
     """ Course serializer. """
     id = serializers.CharField()  # pylint: disable=invalid-name
     name = serializers.CharField(read_only=True)
+    verification_deadline = serializers.DateTimeField(blank=True)
     modes = CourseModeSerializer(many=True, allow_add_remove=True)
 
+    def validate(self, attrs):
+        """ Ensure the verification deadline occurs AFTER the course mode enrollment deadlines. """
+        verification_deadline = attrs.get('verification_deadline', None)
+
+        if verification_deadline:
+            upgrade_deadline = datetime.max.replace(tzinfo=pytz.utc)
+
+            # Find the earliest upgrade deadline
+            for mode in attrs['modes']:
+                expires = mode.expiration_datetime
+                if expires:
+                    upgrade_deadline = min(expires, upgrade_deadline)
```

Comment:
Okay, I'm still concerned that this isn't the correct behavior.  Suppose that there's a professional-with-verification mode configured so that:

* `expiration_datetime` is `None`
* `verification_deadline` is 2017-01-2

Then `upgrade_deadline` is going to default to 9999-01-01 (or something), and we're going to get a validation error.  But in this case, that's what we want to set this to, since (a) we don't want the mode to expire before enrollment closes, and (b) we want a verification deadline so users are prompted to verify before the end of the course.

Is there something I'm not understanding about this case that makes this concern irrelevant?